core/store/src/trie/mod.rs

```diff
@@ -595,7 +719,12 @@ impl Trie {
         }
         let bytes = self.storage.retrieve_raw_bytes(hash)?;
         let node = RawTrieNodeWithSize::decode(&bytes).map_err(|err| {
-            StorageError::StorageInconsistentState(format!("Failed to decode node {hash}: {err}"))
+            StorageError::NotANode(NotANodeError {
+                hash: hash.clone(),
+                len: bytes.len(),
+                data_prefix: format!("{:?}", &bytes[..std::cmp::min(10, bytes.len())]),
+                error: format!("Failed to decode node {hash}.: {err}"),
```

Comment:
Perhaps instead just `err: err` and then Display / Debug can format it however it wants?

Reply:
done.